plugins/woocommerce/src/Blocks/Domain/Services/CheckoutLink.php

```diff
@@ -193,7 +193,7 @@ protected function get_checkout_link() {
 
 		// If the user is logged in, the session is tied to the user ID. Do not use a cart token.
 		if ( ! is_user_logged_in() ) {
-			$session_token = CartTokenUtils::get_cart_token( wc()->session->get_customer_id() );
+			$session_token = CartTokenUtils::get_cart_token( (string) wc()->session->get_customer_id() );
```

Comment:
I did change `get_customer_id()` to return only strings but I left these in place just in case something is extending session handlers and not following suit.